pages/lookuptables.rst

```diff
@@ -0,0 +1,129 @@
+.. _lookuptables:
+
+*************
+Lookup Tables
+*************
+
+Graylog 2.3 introduced the lookup tables feature. It allows you to lookup/map/translate
+message field values into new values and write them into new message fields or
+overwrite existing fields.
+A simple example is to use a static CSV file to map IP addresses to host names.
+
+Components
+----------
+
+The lookup table systems consists of four components.
+
+- Data adapters
+- Caches
+- Lookup tables
+- Lookup results
+
+Data Adapters
+^^^^^^^^^^^^^
+
+Data adapters are used to do the actual lookup for a value. They might read
+from a CSV file, connect to a database or execute HTTP requests to receive
+the lookup result.
+
+Data adapter implementations are pluggable and new ones can be added through plugins.
+
+Caches
+^^^^^^
+
+The caches are responsible for caching the lookup results to improve the lookup
+performance and/or to avoid overloading databases and APIs. They are separate
+entities to make it possible to reuse a cache implementation for different
+data adapters. That way, the data adapters do not have to care about caching
+and do not have to implement it on their own.
+
+Cache implementations are pluggable and new ones can be added through plugins.
+
+Lookup Tables
+^^^^^^^^^^^^^
+
+The lookup table component ties together a data adapter instance and a cache
+instance. It is needed to actually enable the usage of the lookup table in
+extractors, converters, pipeline functions and decorators.
+
+Lookup Results
+^^^^^^^^^^^^^^
+
+The lookup result is returned by a lookup table through the data adapter and
+can contain two types of data. A **single value** and a **multi value**.
+
+The **single value** can be a string, number or boolean and will be used in
+extractors, converters, decorators and pipeline rules. In our CSV example to
+lookup host names for IP addresses, this would be the host name string.
+
+A **multi value** is a map/dictionary-like data structure and can contain
+several different values. This is useful if the data adapter can provide
+multiple values for a key. A good example for this would be the geo-ip data
+adapter which does not only provide the latitude and longitude for an IP
+address, but also information about the city and country of the location.
+Currently, the multi value can only be used in a pipeline rule when using the
+``lookup()`` pipeline function.
+
+**Example 1:** Output for a CSV data adapter including a single value
+and a multi value.
+
+.. image:: /images/lookuptables/single-value-example.png
+
+**Example 2:** Output for the geo-ip data adapter including a single value
+and a multi value.
+
+.. image:: /images/lookuptables/multi-value-example.png
+
+Setup
+-----
+
+The lookup tables can be configured on the "System/Lookup Tables" page.
```

Comment:
in other parts of the Docs we use `System > Lookup Tables` (Code Markup) that should be identical in all parts